runtime/astar/src/lib.rs

```diff
@@ -1517,6 +1521,98 @@ impl pallet_migrations::Config for Runtime {
+                | pallet_democracy::Call::fast_track { .. }
+                | pallet_democracy::Call::emergency_cancel { .. }
+                | pallet_democracy::Call::cancel_referendum { .. }
+                | pallet_democracy::Call::vote { .. }
+                | pallet_democracy::Call::remove_vote { .. }
+                | pallet_democracy::Call::veto_external { .. },
+            )
+            | RuntimeCall::Proxy(_)
+            | RuntimeCall::TxPause(_)
+            | RuntimeCall::SafeMode(_) => true,
+            _ => false,
+        }
+    }
+}
+
+/// Calls that cannot be paused by the tx-pause pallet.
+pub struct TxPauseWhitelistedCalls;
+impl frame_support::traits::Contains<RuntimeCallNameOf<Runtime>> for TxPauseWhitelistedCalls {
+    fn contains(full_name: &RuntimeCallNameOf<Runtime>) -> bool {
+        let pallet_name = full_name.0.as_slice();
+        matches!(
+            pallet_name,
+            b"System"
+                | b"Timestamp"
+                | b"ParachainSystem"
+                | b"Council"
+                | b"TechnicalCommittee"
+                | b"Sudo"
+                | b"TxPause"
+                | b"SafeMode"
+        )
+    }
+}
+
+parameter_types! {
+    // Should be sufficient for the team to react & come up with longer term solution.
+    pub const EnterDuration: BlockNumber = 12 * HOURS;
+    // Extension in case previous time isn't enough.
+    pub const ExtendDuration: BlockNumber = 2 * HOURS;
+    // Permisonless safe mode entry isn't allowed yet.
+    pub const NoPermisionlessEntry: Option<Balance> = None;
+    pub const NoPermisionlessExtension: Option<Balance> = None;
+}
+
+impl pallet_safe_mode::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type Currency = Balances;
+    type RuntimeHoldReason = RuntimeHoldReason;
+    type WhitelistedCalls = SafeModeWhitelistedCalls;
+    type EnterDuration = EnterDuration;
+    type EnterDepositAmount = NoPermisionlessEntry;
+    type ExtendDuration = ExtendDuration;
+    type ExtendDepositAmount = NoPermisionlessExtension;
+    type ForceEnterOrigin =
+        EnsureWithSuccess<EnsureRootOrHalfTechCommitteeOrTwoThirdCouncil, AccountId, EnterDuration>;
+    type ForceExtendOrigin = EnsureWithSuccess<
+        EnsureRootOrHalfTechCommitteeOrTwoThirdCouncil,
+        AccountId,
+        ExtendDuration,
+    >;
+    type ForceExitOrigin = EnsureRootOrHalfTechCommitteeOrTwoThirdCouncil;
+    type ForceDepositOrigin = EnsureRoot<AccountId>;
+    type ReleaseDelay = ();
+    type Notify = DappStaking;
+    type WeightInfo = pallet_safe_mode::weights::SubstrateWeight<Runtime>;
+}
+
+impl pallet_tx_pause::Config for Runtime {
+    type RuntimeEvent = RuntimeEvent;
+    type RuntimeCall = RuntimeCall;
+    type PauseOrigin = EnsureRootOrHalfTechCommitteeOrTwoThirdCouncil;
+    type UnpauseOrigin = EnsureRootOrHalfTechCommitteeOrTwoThirdCouncil;
+    type WhitelistedCalls = TxPauseWhitelistedCalls;
```

Comment:
When I've added tx_pause into Shibuya, I didn't specified whitelisting on purpose, reference: https://github.com/AstarNetwork/Astar/pull/1388#discussion_r1862148779

> In fact, I've just removed the `TxPauseWhitelistedCalls` type and its `Contains` trait implementation. `tx-pause` is more granular, we should be able to pause any call in an emergency.

But I guess you're adding it now on purpose to avoid being stuck?

Reply:
I haven't checked this, but thanks for the reminder.

I've added it since Pierre commented it, and _avoiding being stuck_ just came as an extra reason later.